src/macaron/slsa_analyzer/ci_service/github_actions/analyzer.py

```diff
@@ -0,0 +1,718 @@
+    """Get reachable secrets to a GitHub Actions step.
+
+    Parameters
+    ----------
+    step_node: BashNode
+        The target GitHub Action step node.
+
+    Yields
+    ------
+    str
+        The reachable secret variable name.
+    """
+    job_node = step_node.caller
+    if not isinstance(job_node, GitHubJobNode):
+        return
+
+    def _find_secret_keys(ast: dict | None) -> Iterable[str]:
+        if ast is None:
+            return
+        if ast.get("Env") and ast["Env"].get("Vars"):
+            for key, val in ast["Env"]["Vars"].items():
+                if val.get("Value") is None or val["Value"].get("Value") is None:
+                    continue
+                resolved_val = val["Value"]["Value"]
+                if list(find_expression_variables(value=resolved_val, exp_var="secrets")):
+                    yield key
+
+    # Get reachable secrets set as environment variables in the job.
+    yield from _find_secret_keys(job_node.parsed_obj)
+
+    # Get reachable secrets set as environment variables in the step.
+    if step_node.node_type == BashScriptType.INLINE:
+        yield from _find_secret_keys(step_node.parsed_step_obj)
+
+
+def get_ci_events(workflow_node: GitHubWorkflowNode) -> list[str] | None:
+    """Get the CI events that trigger the GitHub Action workflow.
+
+    Parameters
+    ----------
+    workflow_node: GitHubWorkflowNode
+        The target GitHub Action workflow node.
+
+    Returns
+    -------
+    list[str] | None
+        The list of event names or None.
+    """
+    result = []
+    ast = workflow_node.parsed_obj
+    if ast.get("On") is None:
+        raise GitHubActionsValueError(f"Unable to find `On` event in {workflow_node.source_path} GitHub Action.")
+    for hook in ast["On"]:
+        if hook.get("Hook") is None or hook["Hook"].get("Value") is None:
+            logger.debug("Unable to find event hooks in %s GitHub Action.", workflow_node.source_path)
+            return None
+        result.append(hook["Hook"]["Value"])
+    return result
+
+
+class SetupJava(Language, ThirdPartyAction):
+    """This class models the official setup-java GitHub Action from GitHub.
+
+    For the table of supported distributions see:
+    https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions
+    """
+
+    #: Name of the GitHub Action.
+    action_name = "actions/setup-java"
+
+    #: Version of the GitHub Action.
+    action_version: None
+
+    def __init__(self, external_node: GitHubWorkflowNode):
+        """Initialize the setup-java GitHub Action model.
+
+        Parameters
+        ----------
+        external_node: GitHubWorkflowNode
+            The external GitHub Action workflow node.
+        """
+        step = external_node.parsed_obj
+        self._lang_name = BuildLanguage.JAVA
+        self._lang_distributions = None
+        self._lang_versions = None
+        self._lang_url = "https://github.com/actions/setup-java"
```

Comment:
For language name and language url, do we use these private attributes anywhere? If not, can we return those values as part of `lang_name` and `lang_url` directly?
```
    @property
    def lang_name(self) -> str:
        """Get the name of the language."""
        return BuildLanguage.JAVA

   @property
    def lang_url(self) -> str | None:
        """Get the URL that provides information about the language distributions and versions."""
        return "https://github.com/actions/setup-java"
```

Reply:
That's just an implementation detail. I prefer to set all the properties in one place in the `__init__` function and avoid scattering them around.